edb/server/compiler/compiler.py

```diff
@@ -419,7 +419,8 @@ def _compile_and_apply_migration_command(
                 return self._compile_and_apply_ddl_command(ctx, ddl_plan)
 
             elif isinstance(cmd, s_migrations.GetMigration):
-                ddl_text = s_ddl.ddl_text_from_migration(schema, migration)
+                delta = migration.get_delta(schema)
+                ddl_text = s_ddl.ddl_text_from_delta(schema, delta)
```

Comment:
This actually reminds me about storing migrations in the DB and then we can drop the requirement to always do them in a transaction, right?

Reply:
Possibly. We'll need to review the design once we move on to making the proper migration cli.